include/NovelRT/Graphics/ImageRect.h

```diff
@@ -0,0 +1,52 @@
+// Copyright © Matt Jones and Contributors. Licensed under the MIT Licence (MIT). See LICENCE.md in the repository root for more information.
+
+#ifndef NOVELRT_GRAPHICS_IMAGERECT_H
+#define NOVELRT_GRAPHICS_IMAGERECT_H
+#ifndef NOVELRT_H
+#error Please do not include this directly. Use the centralised header (NovelRT.h) instead!
+#endif
+
+namespace NovelRT::Graphics {
+
+  class ImageRect : public RenderObject {
+    friend class TextRect;
```

Comment:
Why friend rather than making the necessary bits `protected`?

Reply:
I'm unsure how that would achieve the same effect.

TextRect is invoking an `Internal` method on ImageRect. How would `protected` support this?

Reply:
Which internal method?

I'm trying to think about how this could be configured (probably in a future PR) so users could define their own `RenderObject` types if desired (we aren't very extensible if only we can define specialized render objects).

Reply:
there's a private method that allows for the texture to be reassigned in an `ImageRect` with a buffer handle, which only `TextRect` uses. Specifically, `setTextureInternal`.

As for custom `RenderObject` types, there's nothing stopping them inheriting `RenderObject` and implementing them as we've done with our default ones.

Reply:
Right, but users can't (for example) inherit `ImageRect` and create their own `TextRect`.

Reply:
I see, so you reckon by making it `protected` they could also use this internal method for their own extensibility purposes?

Reply:
Right. `friend class` is basically C# `internal` (but with access to `private` members as well) so it tightly couples bits together and prevents external users from doing the same.

In general, it would be good to only use it where really necessary and try to design with `protected` and `public` members instead.

Reply:
okay, but I didn't think `protected` resolves me completely removing `friend class` in this context does it? I thought `protected` meant only this type and inheritors?

Reply:
I was thinking the member should probably be public -or- there should be `TextureRect` class which `ImageRect` inherits from.

There are multiple scenarios where devs may want to set a texture without having a physical image on disk first (including procedural generated data)

Reply:
(also, to clarify, this could be a small follow up PR)

Reply:
Okay, I'll make an issue.

Reply:
Propagated to #100 . Resolving.